MyApplication/app/src/main/java/test/test/com/myapplication/MainActivity.java

```diff
@@ -1,42 +1,118 @@
-        } else if (animationState == FRAGMENT_BACK_ANIMATION) {
-       //     ft.setCustomAnimations(R.animator.fragment_scale_in, R.animator.slide_out_right);
+    @Override
+    public void onRequestPermissionsResult(int requestCode, String permissions[], int[] grantResults) {
+        if (PermissionUtils.isLocationPermissionsApproved(this)) {
+            getLocation();
         }
+    }
+
+    private void setUpSlider() {
+        mSlider.setCoveredFadeColor(0);
+        mSlider.setOverlayed(false);
+        mSlider.setAnchorPoint(SLIDING_PANEL_ANCHOR_POINT);
+        mSlider.setPanelHeight(0);
+        mSlider.addPanelSlideListener(new SlidingUpPanelLayout.PanelSlideListener() {
+            @Override
+            public void onPanelSlide(View panel, final float slideOffset) {
+                updateMapMargins((int) (panel.getHeight() * slideOffset) / 2);
+            }
+
+            @Override
+            public void onPanelStateChanged(View panel, SlidingUpPanelLayout.PanelState previousState, SlidingUpPanelLayout.PanelState newState) {
+
+            }
+        });
+    }
+
+    private void updateMapMargins(int topMargin) {
+
+        FrameLayout.LayoutParams params = (FrameLayout.LayoutParams) mapLayout.getLayoutParams();
+        params.topMargin = -topMargin;
+        params.bottomMargin = topMargin;
+        mapLayout.setLayoutParams(params);
+    }
+
+    public void changeFragment(Fragment fragment, String className) {
+        android.support.v4.app.FragmentTransaction ft = getSupportFragmentManager().beginTransaction();
+        ft.addToBackStack(null);
         ft.replace(R.id.fragment_container, fragment, className).commit();
-//        FragmentTransaction ft  = getSupportFragmentManager().beginTransaction();
-////        fragmentManager.beginTransaction()
-////                .replace(R.id.fragment_container, fragment).commit();
-//
-//        if (animationState == FRAGMENT_ANIMATION) {
-//            ft.setCustomAnimations(R.animator.slide_in_right, R.animator.fragment_scale_out);
-//        } else if (animationState == FRAGMENT_BACK_ANIMATION) {
-//            ft.setCustomAnimations(R.animator.fragment_scale_in, R.animator.slide_out_right);
-//        }
-//        ft.replace(R.id.fragment_container, fragment, className).commit();
+    }
+
+    @Override
+    public void OnRouteReady(String distance) {
+        listener.OnRouteReady1(distance);
+        mSlider.setPanelState(SlidingUpPanelLayout.PanelState.ANCHORED);
+        mSlider.setPanelHeight((int) SLIDING_PANEL_ANCHOR_POINT);
+    }
+
+    @Override
+    public void onBackPressed() {
+        if (mSlider.getPanelState() == SlidingUpPanelLayout.PanelState.ANCHORED || mSlider.getPanelState() == SlidingUpPanelLayout.PanelState.EXPANDED) {
+            mSlider.setPanelState(SlidingUpPanelLayout.PanelState.HIDDEN);
+        } else if (searchFragment != null && searchFragment.isVisible()) {
+            finish();
```

Comment:
finish will close the activity... where do we go from here? 
This is the main activity!